src/box/lua/config/applier/roles.lua

```diff
@@ -184,6 +190,18 @@ local function preload(config)
     roles_state.metadata = {}
     update_roles_metadata(role_names)
 
+    -- Check `fail_if` tag for roles and raise an error if the check fails.
+    for _, role_name in ipairs(role_names) do
+        local md = roles_state.metadata[role_name]
+        if md ~= nil and md['fail_if'] ~= nil then
+            local expr = md['fail_if']
+            if expression.eval(expr, conditional_vars) then
```

Comment:
Parsing error (or other similar error) doesn't report the role name, but it would be nice to narrow down the source of the problem.

The same for the `applier/app.lua`.

Reply:
Added error handling and tests for such cases.